flink-yarn/src/main/scala/org/apache/flink/yarn/ApplicationMasterBase.scala

```diff
@@ -120,18 +125,76 @@ abstract class ApplicationMasterBase {
         config.setInteger(ConfigConstants.JOB_MANAGER_WEB_PORT_KEY, 0)
       }
 
-      val (actorSystem, jmActor, archiveActor, webMonitor) =
+      // we try to start the JobManager actor system using the port definition
+      // from the config.
+      // first, we check if the port is available by opening a socket
+      // if the actor system fails to start on the port, we try further
+      val amPortRange: String = config.getString(ConfigConstants.YARN_APPLICATION_MASTER_PORT,
+        ConfigConstants.DEFAULT_YARN_APPLICATION_MASTER_PORT)
+      val portsIterator = NetUtils.getPortRangeFromString(amPortRange)
+
+      // method to start the actor system.
+      def startActorSystem(portsIterator: java.util.Iterator[Integer]): // return type -> next line
```

Comment:
When you do line breaks in the method header you usually put each parameter in a new line with double indentation and the return type with the colon in a new line with single indentation. Like: 

```
def foo(
    foo: Bar
    bar: Foo)
  : Foobar = {
}
```